vip-cache-manager/vip-cache-manager.php

```diff
@@ -61,6 +64,96 @@ public function clear_queued_purge_urls() {
 		$this->purge_urls = [];
 	}
 
+	/**
+	 * Display a button to purge the cache for the specific URL and its assets
+	 *
+	 * @return void
+	 */
+	public function admin_bar_callback( WP_Admin_Bar $admin_bar ) {
+		if ( is_admin() || ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		$admin_bar->add_menu(
+			[
+				'id'     => 'vip-purge-page',
+				'parent' => null,
+				'group'  => null,
+				'title'  => 'Flush Cache for Page',
+				'href'   => '#',
+				'meta'   => [
+					'title' => 'Flush Page cache for this page and its assets',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Enqueue the button for users who have the needed caps.
+	 *
+	 * @return void
+	 */
+	public function button_enqueue_scripts() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		wp_enqueue_script( 'purge-page-cache-btn', plugins_url( '/js/admin-bar.js', __FILE__ ), [], '1.1', true );
+		wp_localize_script( 'purge-page-cache-btn', 'VIPPageFlush', [
+			'nonce' => wp_create_nonce( 'purge-page' ),
+			'ajaxurl' => add_query_arg( [ 'action' => 'vip_purge_page_cache' ], admin_url( 'admin-ajax.php' ), 
+			),
+		] );
+	}
+
+	/**
+	 * AJAX callback that performs basic security checks and payload validation and queues urls for the purge.
+	 *
+	 * @return void
+	 */
+	public function ajax_vip_purge_page_cache() {
+		$req = json_decode( file_get_contents( 'php://input' ) );
+
+		if ( json_last_error() ) {
+			\Automattic\VIP\Stats\send_pixel( [
+				'vip-go-cache-manager-action' => 'user-url-purge',
+				'vip-go-cache-manager-site'   => VIP_GO_APP_ID,
+				'vip-go-cache-manager-status' => 'bad-payload',
+			] );
+			wp_send_json_error( [ 'error' => 'Malformed payload' ], 400 );
+		}
+
+		if ( ! ( current_user_can( 'manage_options' ) && isset( $req->nonce ) && wp_verify_nonce( $req->nonce, 'purge-page' ) ) ) {
+			\Automattic\VIP\Stats\send_pixel( [
+				'vip-go-cache-manager-action' => 'user-url-purge',
+				'vip-go-cache-manager-site'   => VIP_GO_APP_ID,
+				'vip-go-cache-manager-status' => 'deny',
+			] );
+
+			wp_send_json_error( [ 'error' => 'Unauthorized' ], 403 );
+		}
+
+		$urls = is_array( $req->urls ) && ! empty( $req->urls ) ? $req->urls : [];
+
+		// URLs are validated in queue_purge_url.
+		foreach ( $urls as $url_to_purge ) {
+			$this->queue_purge_url( $url_to_purge );
+		}
+
+		\Automattic\VIP\Stats\send_pixel( [
+			'vip-go-cache-manager-action' => 'user-url-purge',
+			'vip-go-cache-manager-site'   => VIP_GO_APP_ID,
+			'vip-go-cache-manager-status' => 'success',
```

Comment:
A bit more specific naming for the stats:

```suggestion
			'vip-go-cache-manager-url-purge-by-site'   => VIP_GO_APP_ID,
			'vip-go-cache-manager-url-purge-status' => 'success',
```